src/components/ConstructorPage/useNavigation.tsx

```diff
@@ -14,6 +20,13 @@ function findItem(right: NavigationItemModel[], left: NavigationItemModel[], typ
     return right.some((item) => item.type === type) || left.some((item) => item.type === type);
 }
 
+type CustomNavigationData = {
```

Comment:
if I understand correctly, then we have a type from @gravity-ui/page-constructor and in it the parameter `logo` for example is specified as mandatory and we need to make it optional, then I would suggest doing something like this

```
type NavigationDataWithOptionalLogo = Omit<NavigationData, 'logo'> & {
  logo?: NavigationData['logo']
};
```